### ISSUE TITLE
App Crashes on IOS 15 and below

### ISSUE THREAD

**Issue by donpaul120** [pichillilorenzo/flutter_inappwebview#1520]
Each time i call controller.getSettings() the app crashes with this and i have no idea what this means.

```logs
023-02-01 15:20:28.375805+0100 Runner[62348:2123240] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[WKPreferences isTextInteractionEnabled]: unrecognized selector sent to instance 0x60000371f430'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001803f7978 __exceptionPreprocess + 236
	1   libobjc.A.dylib                     0x0000000180188800 objc_exception_throw + 56
	2   CoreFoundation                      0x0000000180406f14 +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
	3   CoreFoundation                      0x00000001803fbbf0 ___forwarding___ + 1408
	4   CoreFoundation                      0x00000001803fde7c _CF_forwarding_prep_0 + 92
	5   flutter_inappwebview                0x000000010552ee30 $s20flutter_inappwebview20InAppWebViewSettingsC07getRealG03objSDySSypSgGAA0cdeF0CSg_tF + 6876
	6   flutter_inappwebview                0x0000000105508a34 $s20flutter_inappwebview12InAppWebViewC11getSettingsSDySSypSgGSgyF + 68
	7   flutter_inappwebview                0x00000001055668d0 $s20flutter_inappwebview22WebViewChannelDelegateC6handle_6resultySo17FlutterMethodCallC_yypSgctF + 16712
	8   flutter_inappwebview                0x00000001054d35dc $s20flutter_inappwebview15ChannelDelegateC7channelACSo013FlutterMethodC0C_tcfcySo0fG4CallC_yypSgctcACcfu_yAH_yAIctcfu0_TA + 44
	9   flutter_inappwebview                0x00000001054d3654 $sSo17FlutterMethodCallCypSgIegn_Ieggg_AByXlSgIeyBy_IeyByy_TR + 108
	10  Flutter                             0x0000000105ec2cd8 __45-[FlutterMethodChannel setMethodCallHandler:]_block_invoke + 112
	11  Flutter                             0x00000001059315fc ___ZN7flutter25PlatformMessageHandlerIos21HandlePlatformMessageENSt3__110unique_ptrINS_15PlatformMessageENS1_14default_deleteIS3_EEEE_block_invoke + 104
	12  libdispatch.dylib                   0x00000001081340b4 _dispatch_call_block_and_release + 24
	13  libdispatch.dylib                   0x0000000108135884 _dispatch_client_callout + 16
	14  libdispatch.dylib                   0x0000000108143f84 _dispatch_main_queue_callback_4CF + 1044
	15  CoreFoundation                      0x0000000180365d1c __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
	16  CoreFoundation                      0x000000018036025c __CFRunLoopRun + 2448
	17  CoreFoundation                      0x000000018035f3bc CFRunLoopRunSpecific + 572
	18  GraphicsServices                    0x000000018afdd70c GSEventRunModal + 160
	19  UIKitCore                           0x00000001843f03d0 -[UIApplication _run] + 964
	20  UIKitCore                           0x00000001843f51ac UIApplicationMain + 112
	21  Runner                              0x000000010436ad38 main + 64
	22  ???                                 0x000000010444de50 0x0 + 4366589520
	23  ???                                 0xfb69800000000000 0x0 + 18116151713562886144
)

libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[WKPreferences isTextInteractionEnabled]: unrecognized selector sent to instance 0x60000371f430'
terminating with uncaught exception of type NSException
```

**Comment by Manuito83** [comment]
I'm getting the same. Crashes on iOS 14.6 and 14.8.X whenever `getSettings()` is called (directly or indirectly, e.g. by calling `getHtml()`). Tried with Flutter 3.3.10 and 3.7.1, and package version beta.1 to beta.22.

**Comment by donpaul120** [comment]
The problem is coming from this line, if you fork it and comment it out it works buh i still don't know why i have to remove it in the first place since `isTextInteractionEnabled` and `upgradeKnownHostsToHTTPS` is available form iOS 14.5 above

<img width="911" alt="Screenshot 2023-02-02 at 15 41 57" src="https://user-images.githubusercontent.com/6430195/216355072-6c49cc8c-4415-4014-8877-6084a9dfb579.png">

cc @Manuito83

**Comment by Manuito83** [comment]
Thanks @donpaul120, this is exactly the same I've done (at least as a temporary fix). I've forked it, but instead of commenting it out, I changed it to a higher version both in `InAppWebView.swift` and `InAppWebViewSettings.swift`. With the platform versions I've got available, it seems that > 15.0 should work (or at least it does in the simulator). This matches what I'm seeing in Crashlytics, where only version 14.6, 14.8.0 and 14.8.1 seem affected.

```
if #available(iOS 15.0, *) {
    realSettings["isTextInteractionEnabled"] = configuration.preferences.isTextInteractionEnabled
}
```
```
if #available(iOS 15.0, *) {
    if newSettingsMap["isTextInteractionEnabled"] != nil && settings?.isTextInteractionEnabled != newSettings.isTextInteractionEnabled {
        configuration.preferences.isTextInteractionEnabled = newSettings.isTextInteractionEnabled
    }
}
```

But, as you said, iOS documentation states that it should be available > 14.5, so I'm not really sure why this is an issue (but I don't think we should submit for this until we find out). In any case, it doesn't look like we are missing much (except the ability to use this particular option in a couple of iOS versions).
